### PR TITLE
fix error function in x64

### PR DESCRIPTION
**Pull request by meepen**
guess when i made that comment back forever ago about advanceluaerrorreporter i didn't think it through enough, here's a fix for errors not displaying at all now

### CODE REVIEW

**Comment by FredyH** [comment]
As I've pointed out in #32 , calling callbacks for queries in transactions is a bad idea, so that needs to go.
I am also fairly certain that referencing the callbacks of queries in transactions somehow led to memory leaks because of a cycle in the references.

File: MySQLOO/source/LuaObjectBase.cpp [hunk 1]
```diff
@@ -153,16 +153,25 @@ void LuaObjectBase::runFunction(GarrysMod::Lua::ILuaBase* LUA, int funcRef, cons
 	va_end(arguments);
 }
 
+static int BuildErrorStack(lua_State *state) {
```
Comment (FredyH) [inline MySQLOO/source/LuaObjectBase.cpp]: please rename to buildErrorStack (lowercase b), to be consistent with other static function names

File: MySQLOO/source/LuaObjectBase.cpp [hunk 2]
```diff
@@ -153,16 +153,25 @@ void LuaObjectBase::runFunction(GarrysMod::Lua::ILuaBase* LUA, int funcRef, cons
...
+	GarrysMod::Lua::ILuaBase* LUA = state->luabase;
+
+	LUA->PushSpecial(GarrysMod::Lua::SPECIAL_GLOB);
+	LUA->GetField(-1, "debug");
```
Comment (FredyH) [inline MySQLOO/source/LuaObjectBase.cpp]: what happens if debug or debug.traceback are nil? I am not sure if we should be required to care for these circumstances but maybe it'd be good to anyways?

**Comment by meepen** [comment]
```
> pluto.db.query("SELECT 1;", nil, function() error "test" end)...
lua_run:1: test
stack traceback:
        [C]: at 0x7bad9020
        [C]: in function 'error'
        lua_run:1: in function 'cb'
        addons/pluto-inv/lua/pluto/sv/db.lua:70: in function <addons/pluto-inv/lua/pluto/sv/db.lua:64>
        [C]: in function 'v'
        lua/includes/modules/hook.lua:84: in function <lua/includes/modules/hook.lua:67>
```

File: MySQLOO/source/LuaObjectBase.cpp (hunk 2 above)
Comment (meepen) [reply MySQLOO/source/LuaObjectBase.cpp]: i've made it check if it's a function, and if it's not it will just return the regular error without stack

Comment (meepen) [reply MySQLOO/source/LuaObjectBase.cpp]: i wish i could do more however i don't think i can't without linking to something that gets lua_Debug stuff.

Comment (FredyH) [reply MySQLOO/source/LuaObjectBase.cpp]: Yeah that's what I had in mind, you should do the same for the debug part but I'm probably just being paranoid here.

Comment (meepen) [reply MySQLOO/source/LuaObjectBase.cpp]: if debug is gone, then i believe default gmod will break

if you really want i can add that, but it's up to you

**Pull request merged**